vimeo-networking/src/main/java/com/vimeo/networking/Search.java

```diff
@@ -231,6 +233,49 @@ public static Facet fromString(String text) {
         return VimeoClient.getInstance().search(queryMap, callback);
     }
 
+    private static final String PARAM_VIDEO_SUGGESTION = "video_count";
+    private static final String PARAM_VOD_SUGGESTION = "ondemand_title_count";
+
+    /**
+     * API access to search suggestions.
+     *
+     * @param query                the text query to base suggestions off of. For best performance, the query should be
+     *                             3 characters or more, however, it works with as little as 1 character. If an empty
+     *                             string is provided, this method will immediately notify the callback of a failure
+     *                             and return null.
+     * @param videoSuggestionCount the number of video suggestions to receive. This determines the maximum number of
+     *                             items in the {@link SuggestionResponse#getVideoSuggestions()} list. This will
+     *                             be ignored if a value less than or equal to 0 is provided.
+     * @param vodSuggestionCount   the number of ondemand suggestions to receive. This determines the maximum number of
+     *                             items in the {@link SuggestionResponse#getOndemandSuggestionList()} list. This will
+     *                             be ignored if a value less than or equal to 0 is provided.
+     * @param callback             the callback to be invoked upon completion of this request
+     * @return a {@link Call} that can be used to cancel this request
+     */
+    @Nullable
+    public static Call<SuggestionResponse> suggest(@NotNull String query, int videoSuggestionCount,
+                                                   int vodSuggestionCount,
+                                                   @NotNull ModelCallback<SuggestionResponse> callback) {
+
+        if (query.isEmpty()) {
+            callback.failure(new VimeoError("Query cannot be empty!"));
+            return null;
+        }
+
+        Map<String, String> queryMap = new HashMap<>();
```

Comment:
Minor perf improvement: could supply a default size of 3 here to eliminate array resizing